Mathlib/Data/Matroid/Loop.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2025 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.Matroid.Circuit
+
+/-!
+# Matroid Loops
+A 'loop' of a matroid `M` is an element `e` satisfying one of the following equivalent conditions:
+* `e ∈ M.closure ∅`;
+* `{e}` is dependent in `M`;
+* `{e}` is a circuit of `M`;
+* no base of `M` contains `e`.
```

Comment:
It would be nice to state this using `TFAE`, and might save you some rewrites in practice

Reply:
Done!